planner/core/planbuilder.go

```diff
@@ -1004,8 +1004,20 @@ func getLatestIndexInfo(ctx sessionctx.Context, id int64, startVer int64) (map[i
 		return nil, false, nil
 	}
 	latestIndexes := make(map[int64]*model.IndexInfo)
-	latestTbl, exist := is.TableByID(id)
```

Comment:
Why the `is` not a `infoschema.TemporaryTableAttachedInfoSchema`?

If we can't guarantee the `sctx.GetInfoSchema()` to the the only way to get the correct `InfoSchema`, there will be many corner cases and ... bugs!

Reply:
Because if `sctx.GetInfoSchema()` is not the only API, there will be many places you have to consider appending the TemporaryTable information.

Reply:
When executing update/delete/select for update, getLatestIndexInfo is used for getting the lasted infoschema instead of the snapshot one. For these write operations, they must read the current records instead of the snapshot, so the infoschema should also be lasted.

Reply:
Now I understand the context...
RC use different schema is a known and not solved issue. https://github.com/pingcap/tidb/issues/21498